gadget/update.go

```diff
@@ -331,17 +349,23 @@ func EnsureVolumeCompatibility(gadgetVolume *Volume, diskVolume *OnDiskVolume, o
 		}
 
 		// start offset mismatch
-		if ds.StartOffset != *gv.Offset {
-			return false, fmt.Sprintf("start offsets do not match (disk: %d (%s) and gadget: %d (%s))",
+		if !IsValidStartOffset(ds.StartOffset, vss, vssIdx) {
```

Comment:
nitpick: maybe this helper should return an error with the min and max attached instead of just the bool?

Reply:
Changed